solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -705,6 +709,8 @@ public void preClose() {
     } finally {
       ExecutorUtil.shutdownAndAwaitTermination(customThreadPool);
     }
+
+    ExecutorUtil.shutdownAndAwaitTermination(fireEventListenersThreadPool);
```

Comment:
Do we perhaps want `shutdownNowAndAwaitTermination()` here? At least some related test failures involve submitted tasks in free-floating Thread blocking searcher init; I think `shutdownAndAwaitTermination()` would track those threads, so not _leak_ them, but might nonetheless deadlock in the same way?

Would it be too cute to shutdown the `fireEventListenersThreadPool` via a task submitted to `customThreadPool`, immediately above? I don't think order should matter with this, so "the sooner the better" might be the way to go (unless there's a reason to delay `fireEventListenersThreadPool` shutdown until after the other close tasks are complete). FWIW I think this use would be in keeping with the spirit of the `customThreadPool`.

Reply:
Yea I think thats fair. I did this in dd022238b06df8614a4bd52b2e6fee199b69c9a0